packages/node/src/methods/state-channel/deposit/controller.ts

```diff
@@ -34,22 +42,34 @@ export default class DepositController extends NodeController {
     const { store, provider } = requestHandler;
     const { multisigAddress, amount } = params;
 
+    params.tokenAddress = params.tokenAddress
+      ? params.tokenAddress
+      : AddressZero;
+
     const channel = await store.getStateChannel(multisigAddress);
 
     if (
       channel.hasAppInstanceOfKind(
-        requestHandler.networkContext.ETHBalanceRefundApp
+        requestHandler.networkContext.CoinBalanceRefundApp
       )
     ) {
       return Promise.reject(CANNOT_DEPOSIT);
     }
 
     const address = await requestHandler.getSignerAddress();
 
-    const balanceOfSigner = await provider.getBalance(address);
+    if (params.tokenAddress !== AddressZero) {
+      const contract = new Contract(params.tokenAddress, ERC20.abi, provider);
+      const balance: BigNumber = await contract.functions.balanceOf(address);
```

Comment:
This could easily fail, catch errors for network issues or chain errors for example what if you get revert

Reply:
Addressed.